builder/googlecompute/config.go

```diff
@@ -443,16 +445,24 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		}
 	}
 
+	if c.ImageArchitecture == "" {
+		c.ImageArchitecture = "x86_64"
+	}
+
 	// used for ImageName and ImageFamily
 	imageErrorText := "Invalid image %s %q: The first character must be a lowercase letter, and all following characters must be a dash, lowercase letter, or digit, except the last character, which cannot be a dash"
+	if !validImageName.MatchString(c.ImageName) {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(imageErrorText, "name", c.ImageName))
+	}
 
 	if len(c.ImageName) > 63 {
 		errs = packersdk.MultiErrorAppend(errs,
 			errors.New("Invalid image name: Must not be longer than 63 characters"))
 	}
 
-	if !validImageName.MatchString(c.ImageName) {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(imageErrorText, "name", c.ImageName))
+	if !(c.ImageArchitecture == "x86_64" || c.ImageArchitecture == "ARM64") {
```

Comment:
I wonder, the APIs are capitalising x86_64 as `X86_64`, does that need to be exactly this, or is it case-insensitive?
If everything's case-sensitive and upper-case, may I suggest we process non-empty image architectures to be uppercase (`strings.ToUpper` should do the trick)
Please feel free to disregard if the API is case-insensitive though!

Reply:
I've added an else to when evaluating the default value, if the user has specified a specific image arch, the string will be auto converted to uppercase in the event it was not properly set.